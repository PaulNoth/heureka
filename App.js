import React from 'react';
import { Provider as PaperProvider, Toolbar, ToolbarContent } from 'react-native-paper';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { default as MaterialCommunityIcon } from 'react-native-vector-icons/MaterialCommunityIcons';
import { default as FeatherIcon } from 'react-native-vector-icons/Feather';
import Swiper from 'react-native-swiper';

export default function App() {
  const [screenIndex, setSceenIndex] = useState(0);

  return (
    <PaperProvider>
      <Toolbar style={{backgroundColor: '#1d8bc9'}}>
        <ToolbarContent
          title={'Heureka'}
        />
        </Toolbar>
      <ToolbarControls screenIndex={screenIndex} scrollTo={this.scrollTo} />
      <Swiper ref='swiper' showsPagination={false} loop={true} bounces={true} onIndexChanged={index => setSceenIndex({ index })} >
        <ProductScreen />
        <ShopScreen />
      </Swiper>
    </PaperProvider>
  );
}

const ToolbarControls = (props) => {
  const iconStyle = { flex: 1, marginTop: 20 };
  const wrapperIconStyle = { flex: 1, alignItems: 'center', justifyContent: 'space-around' };
  const highlightWrapperIconStyle = { ...wrapperIconStyle, borderBottomWidth: 2, borderColor: 'red' }
  const highlightIconColor = "#ffffff";
  const iconColor = "rgba(255, 255, 255, 0.4)";
  const screenIndex = props.screenIndex;
  return (<Toolbar style={{backgroundColor: '#1d8bc9'}}>
    <TouchableOpacity style={screenIndex === 0 ? highlightWrapperIconStyle : wrapperIconStyle} onPressIn={() => props.scrollTo(0)}>
      <View>
        <MaterialCommunityIcon name="cellphone-wireless" size={25} color={screenIndex === 0 ? highlightIconColor : iconColor} style={iconStyle} />
      </View>
    </TouchableOpacity>
    <TouchableOpacity style={screenIndex === 1 ? highlightWrapperIconStyle : wrapperIconStyle} onPressIn={() => props.scrollTo(1)}>
      <View>
        <FeatherIcon name="activity" size={25} color={screenIndex === 1 ? highlightIconColor : iconColor} style={iconStyle} />
      </View>
    </TouchableOpacity>
  </Toolbar>);
};
