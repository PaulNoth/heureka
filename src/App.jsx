import React, { useState, useRef } from 'react';
import { Provider as PaperProvider, Appbar, Button } from 'react-native-paper';
import {
  StyleSheet, View,
} from 'react-native';
import Swiper from 'react-native-swiper';
import ProductScreen from './ProductScreen';
import ShopScreen from './ShopScreen';

const styles = StyleSheet.create({
  buttonColorActive: {
    borderBottomColor: 'white',
    borderBottomWidth: 4,
    borderTopStartRadius: 0,
    borderTopEndRadius: 0,
    borderBottomStartRadius: 0,
    borderBottomEndRadius: 0,
    flex: 1,
  },
  buttonColor: {
    // borderBottomColor: 'white',
    // borderBottomWidth: 4,
    borderTopStartRadius: 0,
    borderTopEndRadius: 0,
    borderBottomStartRadius: 0,
    borderBottomEndRadius: 0,
    flex: 1,
  },
  alignmentButtons: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    padding: 0,
    marginBottom: 0,
  },
});

export default function App() {
  const [screenIndex, setSceenIndex] = useState(0);

  const swiperRef = useRef(null);

  const onMenuClick = (index) => {
    swiperRef.current.scrollBy((screenIndex - index) * -1);
  };

  return (
    <PaperProvider>
      <Appbar.Header>
        <Appbar.Content
          title="Heureka"
          subtitle="Pre potraviny"
        />
      </Appbar.Header>
      <View style={styles.alignmentButtons}>
        <Button compact uppercase={false} icon="magnify" mode="contained" style={screenIndex === 0 ? styles.buttonColorActive : styles.buttonColor} onPress={() => onMenuClick(0)}>Najlacnejší produkt</Button>
        <Button compact uppercase={false} icon="cart-outline" mode="contained" style={screenIndex === 1 ? styles.buttonColorActive : styles.buttonColor} onPress={() => onMenuClick(1)}>Môj nákup</Button>
      </View>
      <Swiper
        ref={swiperRef}
        showsPagination={false}
        loop
        bounces
        onIndexChanged={(index) => setSceenIndex(index)}
      >
        <ProductScreen />
        <ShopScreen />
      </Swiper>
    </PaperProvider>
  );
}
