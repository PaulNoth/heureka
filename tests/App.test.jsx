import React from 'react';
import { render } from 'react-native-testing-library';

import App from '../src/App';

describe('<App/>', () => {
    it('has 1 child', () => {
        const tree = render(<App />).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', () => {
        const tree = render(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
