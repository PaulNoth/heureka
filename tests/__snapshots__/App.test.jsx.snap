// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Open up App.js to start working on your app!
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#f50057",
          "borderRadius": 10,
          "color": "#ffffff",
          "fontFamily": "System",
          "fontSize": 10,
          "fontWeight": "400",
          "height": 20,
          "lineHeight": 20,
          "minWidth": 20,
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 4,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      0
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            Increment counter
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
